sklearn/utils/estimator_checks.py

```diff
@@ -575,15 +582,21 @@ def check_estimator(Estimator, generate_only=False):
     parametrize_with_checks : Pytest specific decorator for parametrizing estimator
         checks.
     """
-    if isinstance(Estimator, type):
+    if Estimator != "deprecated":
+        msg = (
+            "'Estimator' was deprecated in favor of 'estimator' in version 1.1 "
+            "and will be removed in version 1.3."
+        )
+        warnings.warn(msg, FutureWarning)
+        estimator = Estimator
+    if isinstance(estimator, type):
```

Comment:
There is an edge case for `check_estimator()` and one does not pass an estimator. We need to add a check here too and a test:

```python
if estimator is None:
    raise ValueError(...)

if isinstance(estimator, type):
    ...
```